src/lib/stats/client.ts

```diff
@@ -0,0 +1,26 @@
+import type {Socket} from 'socket.io-client';
+import {getConfValue} from '../config.js';
+import {getCpuUsage} from './cpu.js';
+
+type Worker = {
+	jobs: Map<string, number>;
+};
+
+const statsConfig = getConfValue<{interval: number}>('stats');
+
+export const report = async (socket: Socket, jobCount: number) => {
+	const cpuUsage = await getCpuUsage();
+
+	socket.emit('probe:stats:report', {
+		cpu: cpuUsage,
+		jobs: {
+			count: jobCount,
+		},
+	});
+};
+
+export const run = (socket: Socket, worker: Worker) => {
+	setInterval(async () => {
+		await report(socket, worker.jobs.size);
+	}, statsConfig.interval * 1000);
```

Comment:
```suggestion
	setInterval(() => {
		report(socket, worker.jobs.size).catch((error) => { /* TODO: log? */ });
	}, statsConfig.interval * 1000);
```

Similar to the recent probe handler issue - the interval function isn't supposed to return a promise, and doing so makes it seem as if the potential errors were handled, but they aren't (I guess here nothing can really throw but still, this is a bad pattern).

Reply:
That's a stretch. There's no way this method would ever throw an error, and attaching an error handler to it only adds a layer of unecessary complexity.